src/OSPSuite.Core/Domain/Mappers/NeighborhoodCollectionToContainerMapper.cs

```diff
@@ -21,40 +24,72 @@ internal class NeighborhoodCollectionToContainerMapper : INeighborhoodCollection
    {
       private readonly IObjectBaseFactory _objectBaseFactory;
       private readonly INeighborhoodBuilderToNeighborhoodMapper _neighborhoodMapper;
+      private readonly IContainerMergeTask _containerMergeTask;
 
       public NeighborhoodCollectionToContainerMapper(
          IObjectBaseFactory objectBaseFactory,
-         INeighborhoodBuilderToNeighborhoodMapper neighborhoodMapper)
+         INeighborhoodBuilderToNeighborhoodMapper neighborhoodMapper,
+         IContainerMergeTask containerMergeTask)
       {
          _objectBaseFactory = objectBaseFactory;
          _neighborhoodMapper = neighborhoodMapper;
+         _containerMergeTask = containerMergeTask;
       }
 
       public IContainer MapFrom(ModelConfiguration modelConfiguration)
       {
          var (_, simulationBuilder) = modelConfiguration;
-         var moleculeNames = simulationBuilder.AllPresentFloatingMoleculeNames();
 
          var neighborhoodsParentContainer = _objectBaseFactory.Create<IContainer>()
             .WithMode(ContainerMode.Logical)
             .WithName(Constants.NEIGHBORHOODS);
 
-         var startValuesForFloatingMolecules = presentMoleculesCachedByContainerPath(moleculeNames, simulationBuilder);
+         var allSpatialStructureAndMergeBehaviors = simulationBuilder.SpatialStructureAndMergeBehaviors;
+         if (!allSpatialStructureAndMergeBehaviors.Any())
+            return neighborhoodsParentContainer;
 
-         var moleculeNamesCopyProperties = simulationBuilder.AllPresentXenobioticFloatingMoleculeNames();
+         var mapToNeighborhood = mapToNeighborhoodDef(modelConfiguration);
+
+         IReadOnlyList<Neighborhood> mapNeighborhoods(SpatialStructure spatialStructure) =>
+            spatialStructure.Neighborhoods.Select(mapToNeighborhood).Where(x => x != null).ToList();
 
          //we use a cache to ensure that we are replacing neighborhoods defined in multiple structures
-         var neighborhoodCache = new ObjectBaseCache<Neighborhood>();
```

Comment:
Funny enough, I had not implemented a merge behavior for neighborhood. Only for spatial structure. Now I have extracted the merge logic out into the ContainerMergeTask